File: InMotionGIT.Common - Copy/DataType/LookUpNumericValue.vb
Imports System.Runtime.Serialization
Imports System.Xml.Serialization

Namespace DataType

    <DataContract(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.Common.DataType")>
    <XmlRoot(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Attributes.TypeStructure("Code")>
    Public Class LookUpNumericValue

        <DataMember()>
        <XmlAttribute()>
        Public Property Code As Integer

        <DataMember()>
        <XmlAttribute()>
        Public Property Description As String

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal code As Integer)
            Me.Code = code
        End Sub

        Public Sub New(ByVal code As Integer, ByVal description As String)
            Me.Code = code
            Me.Description = description
        End Sub

        Public Overrides Function ToString() As String
            Return Description
        End Function

        Public Shared Narrowing Operator CType(ByVal value As LookUpNumericValue) As Integer
            If IsNothing(value) Then
                Return 0
            Else
                Return value.Code
            End If
        End Operator

        Public Shared Widening Operator CType(ByVal code As Integer) As LookUpNumericValue
            Return New LookUpNumericValue(code)
        End Operator

        Public Shared Function LoadFromDataTable(tableInformation As DataTable)
            Dim Result As New List(Of LookUpNumericValue)
            Dim item As LookUpNumericValue
            For Each row As Data.DataRow In tableInformation.Rows
                item = New LookUpNumericValue With {.Code = row("CODE"), .Description = row("DESCRIPTION")}
                Result.Add(item)
            Next
            Return Result
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/DataType/LookUpNumericValue.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Runtime.Serialization;
using System.Xml.Serialization;
using Microsoft.VisualBasic.CompilerServices;

namespace InMotionGIT.Common.DataType
{

    [DataContract(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Serializable()]
    [XmlType(Namespace = "urn:InMotionGIT.Common.DataType")]
    [XmlRoot(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Attributes.TypeStructure("Code")]
    public class LookUpNumericValue
    {

        [DataMember()]
        [XmlAttribute()]
        public int Code { get; set; }

        [DataMember()]
        [XmlAttribute()]
        public string Description { get; set; }

        public LookUpNumericValue() : base()
        {
        }

        public LookUpNumericValue(int code)
        {
            Code = code;
        }

        public LookUpNumericValue(int code, string description)
        {
            Code = code;
            Description = description;
        }

        public override string ToString()
        {
            return Description;
        }

        public static explicit operator int(LookUpNumericValue value)
        {
            if (value == null)
            {
                return 0;
            }
            else
            {
                return value.Code;
            }
        }

        public static implicit operator LookUpNumericValue(int code)
        {
            return new LookUpNumericValue(code);
        }

        public static object LoadFromDataTable(DataTable tableInformation)
        {
            var Result = new List<LookUpNumericValue>();
            LookUpNumericValue item;
            foreach (DataRow row in tableInformation.Rows)
            {
                item = new LookUpNumericValue() { Code = Conversions.ToInteger(row["CODE"]), Description = Conversions.ToString(row["DESCRIPTION"]) };
                Result.Add(item);
            }
            return Result;
        }

    }

}